src/fn/vssop8.ts

```diff
@@ -0,0 +1,88 @@
+import type { AnySoupElement, PcbSilkscreenPath } from "circuit-json"
+import { z } from "zod"
+import { rectpad } from "../helpers/rectpad"
+import { silkscreenRef, type SilkscreenRef } from "src/helpers/silkscreenRef"
+import { length } from "circuit-json"
+
+export const vssop8_def = z.object({
+  fn: z.string(),
+  num_pins: z.literal(8).default(8),
+  w: z.string().default("2.3mm"),
+  h: z.string().default("3.0mm"),
+  pl: z.string().default("0.3mm"),
+  pw: z.string().default("0.25mm"),
+  pad_spacing: z.string().default("0.65mm"),
+})
+
+export const vssop8 = (
+  raw_params: z.input<typeof vssop8_def>,
+): { circuitJson: AnySoupElement[]; parameters: any } => {
+  const parameters = vssop8_def.parse(raw_params)
+
+  // Define silkscreen reference text
+  const silkscreenRefText: SilkscreenRef = silkscreenRef(
+    0,
+    length.parse(parameters.h) / 2 + 0.5, // Adjusted to move above the yellow box
+    0.3,
+  )
+
+  // Define silkscreen boundary
+  const silkscreenLine: PcbSilkscreenPath = {
+    type: "pcb_silkscreen_path",
+    layer: "top",
+    pcb_component_id: "",
+    route: [
+      { x: -length.parse(parameters.w) / 2, y: length.parse(parameters.h) / 2 },
+      { x: length.parse(parameters.w) / 2, y: length.parse(parameters.h) / 2 },
+      { x: length.parse(parameters.w) / 2, y: -length.parse(parameters.h) / 2 },
+      {
+        x: -length.parse(parameters.w) / 2,
+        y: -length.parse(parameters.h) / 2,
+      },
+      { x: -length.parse(parameters.w) / 2, y: length.parse(parameters.h) / 2 },
+    ],
+    stroke_width: 0.1,
+    pcb_silkscreen_path_id: "",
+  }
+
+  return {
+    circuitJson: vssop8Pads(parameters).concat(
+      silkscreenLine as AnySoupElement,
+      silkscreenRefText as AnySoupElement,
+    ),
+    parameters,
+  }
+}
+
+// Get coordinates for VSSOP-8 pads
+export const getVssop8Coords = (parameters: {
```

Comment:
Notice, this should NOT be plural- a coordinate is a single x/y obj, coordinates would be an array

```suggestion
export const getVssop8PadCoord = (parameters: {
```